src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -171,15 +179,15 @@ public void finish() {
      *         output buffer
      */
     @Override
-    public int deflate(byte[] b, int off, int len )  throws NullPointerException, IllegalArgumentException {
+    public int deflate(byte[] b, int off, int len) {
         if(b == null) {
-            throw new NullPointerException("Input is null");
-        }
-        if(len <= 0) {
-            throw new NullPointerException("Input buffer length is less or equal zero.");
+            throw new NullPointerException();
```

Comment:
Why in this change and many more do you remove the txt straight for exceptions?   More info is good.

Reply:
Also for these test changes how does it take to run all the new "single test" unit tests?   Is it much longer then doing it in one test per class with many sub actions?

Reply:
I restored the exception messages. Try-catch blocks were removed as now tests depend on TestNG functionalities. It's much cleaner like that. The test framework is expecting a particular exception that is declared in the annotation tag. These are very quick tests and they do not increase the execution time. Moreover good unit testing practices say that each unit test should test only one thing.  E.g. if you have 10 sequential checks in a single test function and the first check fails, the following 9 won't be checked. If the unit tests fail you want to get a full overview of what is not working, not the first test that failed.